Switch to class component syntax and add NamesList

jtessler opened the pull request:
See individual commits for more details.

AndrewVorwald committed:
Update example data

Use constructor instead of object literal.

This commit is a cherry-pick from AndrewVorwald's fork [1].

[1] https://github.com/AndrewVorwald/family-tree/commit/e889cdfb433d1403538c7c8ad5ee22c8555c7c0c

jtessler committed:
Updated dependencies with 'npm audit fix'

jtessler committed:
Convert Components to class syntax

Currently all starter code Components use the single-function syntax and
the 'useState' hook [1]. This differs from our lessons and sample code,
which unnecessarily confuses students.

[1] https://reactjs.org/docs/hooks-state.html

jtessler committed:
Clean up deeply nested nodes and syntax usage

Now that we are using class syntax for Components, we should encourage
students to define and call new methods rather than nesting more
JSX code in the main render call.

Students aren't familiar with advanced syntax like the ternary operator
and template strings. This change attempts to make the code more
readable for such a person.

jtessler committed:
Add NamesList component

NamesList simply renders a given list of string names into an unordered
list. The list is defined as a constant in App and passed through
InfoBox.

Students implemented this feature in wildly different ways. This commit
marks a new starter code from which all students can branch again.

dss-fed commented on src/NamesList.js:
```diff
@@ -0,0 +1,19 @@
+class NamesList extends React.Component {
+  render() {
+    let namesLiList = [];
+    for (let i in this.props.namesList) {
```
`for ... of` might be a little nicer here.

jtessler replied:
Good suggestion, but unfortunately I'm going off the slides students saw. Code like this will eventually be replaced by `map` once we introduce that method. I'll make a note to introduce the `for...of` syntax though, thanks!

jtessler committed:
Add 'corejs' package, update others

Repl.it depends on 'corejs' for whatever reason. Adding it here so it
loads without any manual intervention.

jtessler committed:
Use 'jsx' extension for all React components

Also adds extensions to import statements to make it extra clear from
where the imports originate.

jtessler committed:
Convert example_data indentation to 2 spaces

jtessler committed:
Revert "Use 'jsx' extension for all React components"

Apparently Repl.it assumes 'App.js' exists in a React project and
doesn't cleanly handle the 'jsx' extension. This change reverts
everything back to a 'js' extension for consistency.

This reverts commit 79ec9ac0a27e5796b1c1cfe088aeb77b203418c0.

jtessler committed:
Add missing key attribute to NamesList
